### PR TITLE
Add methods to enable/disable identity

### PR DESCRIPTION
**Pull request by shekhar-cis**
Add methods to enable & disable the identity.

### CODE REVIEW

File: src/bmconfigparser.py [hunk 1]
```diff
@@ -154,6 +154,16 @@ def validate_bitmessagesettings_maxoutboundconnections(value):
             return False
         return True
 
+    def disable_address(self, address):
+        """"To disable the Address"""
+        self.set(str(address), 'enabled', 'false')
```
Comment (PeterSurda) [inline src/bmconfigparser.py:159]: any error handling?

File: src/bmconfigparser.py [hunk 2]
```diff
@@ -154,6 +154,16 @@ def validate_bitmessagesettings_maxoutboundconnections(value):
...
+        self.save()
+
+    def enable_address(self, address):
+        """"To enable the Address"""
+        self.set(address, 'enabled', 'true')
```
Comment (PeterSurda) [inline src/bmconfigparser.py:164]: also perhaps some error handling

File: src/bmconfigparser.py [hunk 3]
```diff
@@ -10,6 +10,8 @@
 from six import string_types
 from six.moves import configparser
 
+import shared
```
Comment (PeterSurda) [inline src/bmconfigparser.py:13]: import the function directly

File: src/bmconfigparser.py [hunk 4]
```diff
@@ -154,6 +156,30 @@ def validate_bitmessagesettings_maxoutboundconnections(value):
             return False
         return True
 
+    @staticmethod
+    def search_addresses(address, searched_text):
```
Comment (PeterSurda) [inline src/bmconfigparser.py:160]: - add a test
- convert into method

File: src/bmconfigparser.py [hunk 5]
```diff
@@ -154,6 +156,30 @@ def validate_bitmessagesettings_maxoutboundconnections(value):
...
+        """Return the searched label of MyAddress"""
+        return [x for x in [BMConfigParser().get(address, 'label').lower(),
+                address.lower()] if searched_text in x]
+
+    def disable_address(self, address):
```
Comment (PeterSurda) [inline src/bmconfigparser.py:165]: add a test

File: src/bmconfigparser.py [hunk 6]
```diff
@@ -154,6 +156,30 @@ def validate_bitmessagesettings_maxoutboundconnections(value):
...
+        """"To disable the Address"""
+        try:
+            self.set(str(address), 'enabled', 'false')
+            self.save()
+            shared.reloadMyAddressHashes()
+        except (KeyError, TypeError) as e:
+            raise configparser.NoSectionError(e)
+
+    def enable_address(self, address):
```
Comment (PeterSurda) [inline src/bmconfigparser.py:174]: test

**Review by PeterSurda — CHANGES_REQUESTED** [review]

File: src/bmconfigparser.py (hunk 6 above)
Comment (shekhar-cis) [reply src/bmconfigparser.py:172]: Done

File: src/bmconfigparser.py (hunk 4 above)
Comment (shekhar-cis) [reply src/bmconfigparser.py]: Converted into method

File: src/bmconfigparser.py (hunk 2 above)
Comment (PeterSurda) [reply src/bmconfigparser.py]: I changed my mind.

File: src/tests/test_config_address.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        self.enabled.setTemp('enabled_address', 'enabled', 'true')
+        self.assertEqual(self.enabled.safeGet('enabled_address', 'enabled'), 'true')
+
+    def test_enable_disabled_identity(self):
+        """Test enabling the Disabled identity"""
+        self.enabled.setTemp('disabled_address', 'enabled', 'true')
+        self.assertEqual(self.enabled.safeGet('disabled_address', 'enabled'), 'true')
+
+    # def test_enable_non_existent_identity(self):
+    #     self.assertRaises(configparser.NoSectionError, self.nonexistent.setTemp('non_existent_address', 'enabled'))
```
Comment (PeterSurda) [inline src/tests/test_config_address.py:58]: 2 options, inline and context manager:
```
self.assertRaises(configparser.NoSectionError,
                              self.config.enable_address,
                              'non_existent_address')
```
```
with assertRaises(configparser.NoSectionError):
    self.config.enable_address('non_existent_address')
```

File: src/tests/test_config_address.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        self.config = BMConfigParser()
+        self.enabled = self.config
+        self.enabled.read_file(StringIO(enabled_address))
+        self.disabled = self.config
+        self.disabled.read_file(StringIO(disabled_address))
+        self.nonexistent = self.config
+
+    def test_enable_enabled_identity(self):
+        """Test enabling already enabled identity"""
+        self.enabled.setTemp('enabled_address', 'enabled', 'true')
```
Comment (PeterSurda) [inline src/tests/test_config_address.py:49]: `enable_address`, not `setTemp`

**Review by PeterSurda — CHANGES_REQUESTED** [review]

**Review by PeterSurda — APPROVED** [review]

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**